examples/apps/tree-comparison/src/model/newTreeInventoryList.ts

```diff
@@ -177,28 +216,15 @@ export class NewTreeInventoryList extends DataObject implements IInventoryList {
 
 	private makeInventoryItemFromInventoryItemNode(
 		inventoryItemNode: InventoryItemNode,
-	): InventoryItem {
-		const setQuantity = (newQuantity: number) => {
-			// REV: This still seems surprising to me that it's making the remote change, I think
-			// it would be more apparent as .setContent() rather than using the setter.
-			inventoryItemNode.boxedQuantity.content = newQuantity;
-		};
-		const deleteItem = () => {
+	): NewTreeInventoryItem {
+		const removeItemFromTree = () => {
 			// REV: Is this the best way to do this?  Was hoping for maybe just an inventoryItemNode.delete().
```

Comment:
You're doing it the right way, though in the newest API layer the `parentField` property will be going away. It will be replaced with `parent` (a _node_, not a _field_), and you won't be able to get the indix. Instead, you can use `indexOf` like you would in a normal JS array. Don't worry about O(N), we'll make it O(1) under the hood.

We don't currently have `iventoryItemNode.delete()`. Deleting things in lists must be done via the list. In general, modifications of things happen via the parent of the thing being modified, rather than via the modified thing itself.

Reply:
Is there a way I can at least avoid requiring the list reference?  I'm drawing parallels to DOM APIs - e.g. considering the difference between:
1. `(child) => child.remove()`
2. `(child) => child.parentNode?.removeChild(child)`
3. `(parent, child) => parent.removeChild(child)`

For 1 & 2 I can delegate the responsibility of deletion to someone who only knows the child (1 obv preferable), whereas 3 I have to start handing around the list reference as well.

Maybe `inventoryItemNode.parent.removeAt(inventoryItemNode.parent.indexOf(inventoryItemNode))` would be kind of similar to 2 above?

Reply:
What you did (getting the parent) is fine, when you know the parent is a list. With the latest API, you can use the `is` helper. 
`import { node } from ...tree2`
`if (node.is(node.parent(inventoryItemNode), listSchema)) {...}`